client/services/code_preprocessors/CodePreprocessorDispatcherService.js

```diff
@@ -22,13 +22,32 @@ tie.factory('CodePreprocessorDispatcherService', [
   function(PythonCodePreprocessorService, LANGUAGE_PYTHON) {
     return {
       preprocessCode: function(
-          language, code, mainFunctionName, correctnessTests) {
+          language, code, mainFunctionName, correctnessTests,
+           performanceTests) {
         if (language === LANGUAGE_PYTHON) {
           return PythonCodePreprocessorService.preprocessCode(
+            code, mainFunctionName, correctnessTests, performanceTests);
+        } else {
+          throw Error('Language not supported: ' + language);
+        }
+      },
+      preprocessCorrectnessCode: function(
```

Comment:
I don't understand why we need these extra functions at this level. Shouldn't they be subsidiary to preprocessCode and only added in the PythonCodePreprocessorService? In the grand scheme of things, we are processing the entire code just once and running that preprocessed code just once.